src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/SetAzureDataFactoryIntegrationRuntimeCommand.cs

```diff
@@ -163,7 +169,13 @@ public override void ExecuteCmdlet()
                     }
                     else
                     {
-                        resource.Properties = new SelfHostedIntegrationRuntime();
+                        var selfHosted = new SelfHostedIntegrationRuntime();
+                        if (AuthKey != null)
+                        {
+                            selfHosted.LinkedInfo = new LinkedIntegrationRuntimeKey(new SecureString(AuthKey));
```

Comment:
@zhangyd2015 any reason that we can't just make the `AuthKey` parameter of type `SecureString`?

Reply:
Change the type of AuthKey to SecureString.